gobblin-metrics/src/main/java/gobblin/metrics/graphite/GraphiteReporter.java

```diff
@@ -0,0 +1,296 @@
+        Long.toString(snapshot.getMax()),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.MEAN.getName()),
+        Double.toString(snapshot.getMean()),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.STDDEV.getName()),
+        Double.toString(snapshot.getStdDev()),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.MEDIAN.getName()),
+        Double.toString(snapshot.getMedian()),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.PERCENTILE_75TH.getName()),
+        Double.toString(snapshot.get75thPercentile()),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.PERCENTILE_95TH.getName()),
+        Double.toString(snapshot.get95thPercentile()),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.PERCENTILE_98TH.getName()),
+        Double.toString(snapshot.get98thPercentile()),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.PERCENTILE_99TH.getName()),
+        Double.toString(snapshot.get99thPercentile()),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.PERCENTILE_999TH.getName()),
+        Double.toString(snapshot.get999thPercentile()),
+        timeStamp);
+  }
+
+  private void reportedMetered(MetricContext context, String name, Metered metered, long timeStamp)
+      throws IOException {
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.COUNT.getName()),
+        Long.toString(metered.getCount()),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.RATE_1MIN.getName()),
+        Double.toHexString(convertRate(metered.getOneMinuteRate())),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.RATE_5MIN.getName()),
+        Double.toString(convertRate(metered.getFiveMinuteRate())),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.RATE_15MIN.getName()),
+        Double.toString(convertRate(metered.getFifteenMinuteRate())),
+        timeStamp);
+    this.graphiteSender.send(
+        MetricRegistry.name(context.getName(), name, Measurements.MEAN_RATE.getName()),
+        Double.toString(convertRate(metered.getMeanRate())),
+        timeStamp);
+  }
+
+  private void reportTimer(MetricContext context, String name, Timer timer, long timeStamp) throws IOException {
+    Snapshot snapshot = timer.getSnapshot();
```

Comment:
Most of this code is repeated from reportHistogram. You can instead implement reportSnapshot, and have both Timer and Histogram use this method.